subsys/net/l2/ppp/ppp_l2.c

```diff
@@ -457,4 +496,34 @@ void net_ppp_init(struct net_if *iface)
 
 	k_work_reschedule(&ctx->startup,
 			  K_MSEC(CONFIG_NET_L2_PPP_DELAY_STARTUP_MS));
+
+	if (!tx_init) {
+		k_tid_t tid;
+
+		tx_init = true;
+
+		k_fifo_init(&tx_queue);
+
+		tid = k_thread_create(&tx_handler_thread, tx_handler_stack,
+				      K_KERNEL_STACK_SIZEOF(tx_handler_stack),
+				      (k_thread_entry_t)tx_handler,
+				      NULL, NULL, NULL,
+				      THREAD_PRIORITY, 0, K_FOREVER);
```

Comment:
Both FIFO and thread could be created using `K_FIFO_DEFINE` and `K_THREAD_DEFINE`, so that there is nothing more to do in runtime.

Reply:
> Both FIFO and thread could be created using `K_FIFO_DEFINE` and `K_THREAD_DEFINE`, so that there is nothing more to do in runtime.

Yes, they could but I preferred to do it this way.

Reply:
50% of the code would be dropped with K_FIFO_DEFINE + K_THREAD_DEFINE and additionally we would get rid of `static bool tx_init;` and
```
if (!tx_init) {
		k_tid_t tid;

		tx_init = true;
```
which I am not a fan of. Maybe it would be better to add new `SYS_INIT()` to get rid of that case? `net_ppp_init()` should be reserved for interface initialization IMO, not a static and shared (across multiple interfaces) TX thread.

Reply:
Ah, you have a point here. I will change the code.